packages/auth/src/authClients/clerk.ts

```diff
@@ -0,0 +1,27 @@
+import { Clerk } from '@clerk/clerk-js'
+import { UserResource as ClerkUserResource } from '@clerk/types'
+
+import type { AuthClient } from '.'
+
+export type AuthClientClerk = AuthClient
+
+export type { Clerk }
+
+export type ClerkUser = ClerkUserResource & { roles: string[] | null }
+
+export const clerk = (client: Clerk): AuthClientClerk => {
+  return {
+    type: 'clerk',
+    client,
+    login: async (options?) => client.openSignIn(options || {}),
+    logout: async () => new Promise((res) => client.signOut(res)),
+    signup: async (options?) => client.openSignUp(options || {}),
+    // Clerk uses the session ID PLUS the __session cookie.
+    getToken: async () => client.session.id,
```

Comment:
@dthyresson Sorry about this - it's a confusing Clerk quirk that these two different values have such similar names. I tried to clarify in the comment here but let me know if it needs expansion.

TLDR; Clerk uses an unusual two-session authentication mechanism. It has two keys: the `__session` cookie, which maintains the overall session on the device, and the `session.id` value which maintains state information within that session. It would have been great if they'd picked different names for them in the API but here we are 😓 

I somewhat shoehorned the session.id parameter - which Clerk recommends you [send as a GET parameter](https://docs.clerk.dev/get-started/quickstarts/nextjs#i-add-a-fetcher-utility) - into the Redwood token since I could then rely on the auth system y'all have built to send it over without trying to pipe it over to the API client separately.